tests/test_olio_grid_functions.py

```diff
@@ -0,0 +1,306 @@
+
+
+def test_resequence_nexus_corp():
+
+    # --------- Arrange ----------
+    corner_points = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[10., -10., 0.], [0., -10., 0.]]],
+                                 [[[0., 0., 10.], [10., 0., 10.]], [[10., -10., 10.], [0., -10., 10.]]]]]]])
+
+    expected_result = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[0., -10., 0.], [10., -10., 0.]]],
+                                   [[[0., 0., 10.], [10., 0., 10.]], [[0., -10., 10.], [10., -10., 10.]]]]]]])
+
+    # --------- Act ----------
+    resequence_nexus_corp(corner_points = corner_points)
+
+    # --------- Assert ----------
+    np.testing.assert_almost_equal(corner_points, expected_result)
+
+
+def test_random_cell(tmp_path):
+
+    # --------- Arrange----------
+    epc = os.path.join(tmp_path, 'grid.epc')
+    model = rq.new_model(epc)
+
+    # create a basic block grid
+    dxyz = (10.0, 10.0, 100.0)
+    grid = grr.RegularGrid(model,
+                           extent_kji = (3, 2, 2),
+                           title = 'grid_1',
+                           origin = (0.0, 10.0, 1000.0),
+                           dxyz = dxyz,
+                           as_irregular_grid = True)
+
+    grid_points = grid.points_ref(masked = False)
+
+    # pinch out cells in the k == 1 layer
+    for j in range(3):
+        for i in range(3):
+            grid_points[2][j][i][-1] = 1100.
+    grid.write_hdf5()
+    grid.create_xml(add_cell_length_properties = True)
+    grid_uuid = grid.uuid
+    model.store_epc()
+
+    # collapse cellkji0 == 0,0,0 in the j direction # TODO: figure out why this does not work!
+    for k in [0, 1]:
+        for i in [0, 1]:
+            grid_points[k][1][i][1] = 10.  # same as origin y value
+
+    # check that the grid can be read
+    model = rq.Model(epc)
+    grid_reloaded = grr.any_grid(model, uuid = grid_uuid)
+    corner_points = grid_reloaded.corner_points()
+
+    # reshape corner get the extent of the new grid
+    corner_points_reshaped = corner_points.reshape(1, 1, 12, 2, 2, 2, 3)
+    new_extent = determine_corp_extent(corner_points = corner_points_reshaped)
+
+    # --------- Act----------
+    # call random_cell function 50 times
+    trial_number = 0
+    while trial_number < 50:
+        (k, j, i) = random_cell(corner_points = corner_points)
+        # --------- Assert----------
+        assert 0 <= k < 3
+        assert 0 <= j < 2
+        assert 0 <= i < 2
+        # check that none of the k,j,i combinations that correspond to pinched cells are chosen by the random_cell function
+        assert (k, j, i) not in [(1, 0, 0), (1, 0, 1), (1, 1, 0), (1, 1, 1)]
+        # TODO: figure out why is cell kji0 == (0, 0, 0) still returned?
+        # print(k, j, i)
+        trial_number += 1
+    np.testing.assert_almost_equal(new_extent,
+                                   np.array([3, 2, 2
+                                            ]))  #TODO: confirm this makes sense if an entire k==1 layer is pinched out
```

Comment:
Yes, a pinched out layer still exists as a layer in the logical KJI space, so the grid extent is unchanged.